process_linux.go

```diff
@@ -236,5 +236,14 @@ func (p *initProcess) createNetworkInterfaces() error {
 }
 
 func (p *initProcess) signal(s os.Signal) error {
-	return p.cmd.Process.Signal(s)
+	return maybeSyscall(p.cmd.Process, s)
+}
+
+func maybeSyscall(process *os.Process, s os.Signal) error {
+	if sys, ok := s.(syscall.Signal); ok {
+		// Prefer syscall.Kill because os.Process.Signal masks ESRCH
+		return syscall.Kill(process.Pid, sys)
+	} else {
+		return process.Signal(s)
+	}
```

Comment:
The only safety checks I see is that ensure that the PID is not 0 or 1.  I think in libcontainer we should be pretty safe here as I can't think of a flow of how we could have launched a process with PID 0 or 1 short of the caller setting the PID value in the struct.